api-jersey/src/main/scala/com/yahoo/maha/api/jersey/JsonStreamingOutput.scala

```diff
@@ -169,7 +174,7 @@ case class JsonStreamingOutput(resultList: IndexedSeq[CuratorResult],
           jsonGenerator.writeObject(row.getValue(i))
           i+=1
         }
-        if (rowCountOption.isDefined) {
+        if (includeRowCount) {
```

Comment:
This is also wrong, it only needs to be injected if multi engine query hence check on rowCountOption.